sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedSource.java

```diff
@@ -66,7 +66,12 @@
     if (partitions.isEmpty()) {
       try (Consumer<?, ?> consumer = spec.getConsumerFactoryFn().apply(spec.getConsumerConfig())) {
         for (String topic : spec.getTopics()) {
-          for (PartitionInfo p : consumer.partitionsFor(topic)) {
+          List<PartitionInfo> partitionInfoList = consumer.partitionsFor(topic);
```

Comment:
I'm wondering whether it's possible that we get empty list here since KafkaConsumer javadoc doesn't state it explicitly.

Reply:
Well, I checked it against local Kafka cluster with `auto.create.topics.enable=false`. When I try to read from non-existing topic then `consumer.partitionsFor(topic)` will return null in this case and we have NPE.

Reply:
Thanks for the explanation, Alexey! 
Should we also consider throwing exceptions when we get empty partitions list? Otherwise the pipeline will idle forever.

Reply:
Thanks for the explanation, Alexey! 
Should we also consider throwing exceptions when we get empty partitions list? Otherwise the pipeline will idle forever.

Reply:
We do this a bit later [here](https://github.com/apache/beam/blob/0af8c3bbe8bc4d21dab6b7b45172be900e90620b/sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedSource.java#L81) or you mean something else?

Reply:
ah yeah, that's what I was asking for. Thanks for the pointer!